File: 061_Rotate_List/061_1.java
public class ListNode 
{
    int val;
    ListNode next;
        ListNode(int x) { val = x; }
}

class Solution
{
    public ListNode rotateRight(ListNode head, int k)
    {
        if (head == null || k == 0) return head;
        ListNode current = head;
        int count = 1;
        while (current.next != null)
        {
            current = current.next;
            count++;
        }

        current.next = head;
        for (int i = 0; i < (count - k % count); i++)
            current = current.next;
        head = current.next;
        current.next = null;
        return head;
    }
}

File: 061_Rotate_List/061_1.py
class Solution:
    def rotateRight(self, head, k):
        if not head or not head.next:
            return head
        val1 = head
        count = 1
        while val1.next:
            val1 = val1.next
            count += 1
        val1.next = head
        for _ in range(count - k % count):
            val1 = val1.next
        head = val1.next
        val1.next = None
        return head